### PR TITLE
CI/CD Upgrade.

### PR DESCRIPTION
**Pull request by pwei1018**
*Issue #:* https://github.com/bcgov/entity/issues/7578###

*Description of changes:*
1. Deprecate Jenkins pipeline;
2. Handle test and prod CD flow by Makefile and GHA;
3. Enable Github environments for each repo;
4. Using multiple steps check in GHA CI flow;

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: k8s/README.md
Comment (severinbeauvais) [inline k8s/README.md:9]: Is this an empty file now -- can it just be deleted?

**Review by severinbeauvais — APPROVED** [review]
LGTM but wait for @kialj876 's review before merge please.

**Review by kialj876 — APPROVED** [review]

**Pull request merged**
